src/service/game.go

```diff
@@ -14,31 +14,79 @@ import (
 func NewGameService(r repository.Repository, rand RandomService, max int) GameService {
 	v := validator.New()
 	return &service{
-		Repo: r,
-		Validate: v,
+		Repo:         r,
+		Validate:     v,
 		MaximumCards: max,
-		Rand: rand,
+		Rand:         rand,
 	}
 }
 
-// GameService is interface with methods to interact with redis db	
+// GameService is interface with methods to interact with redis db
 type GameService interface {
-    NewGame(input *TeamInput) (string, error)
+	NewGame(input *TeamInput) (string, error)
 	GetGame(gameID string) (game *Game, err error)
 	UpdateGame(gameID string, input *GameInput) (*Game, error)
 
 	SaveCard(gameID string, card *CardInput) (string, error)
 	GetRandomCard(gameID string) (card *Card, err error)
+<<<<<<< HEAD
+	MakeRandomCards(gameID string, numCards int) (game *Game, err error)
+	MarkCardUsed(gameID, cardID string) error
+=======
 	MarkCardUsed(gameID, cardID string) (*Game, error)
+>>>>>>> f9d1865babbc83d1f03dc600a9a800e9cb036b61
 	ResetGame(gameID string) (*Game, error)
 	DeleteCards(gameID string) error
 }
 
 type service struct {
-	Repo 			repository.Repository
-	Validate 		*validator.Validate
-	MaximumCards 		int
-	Rand 			RandomService
+	Repo         repository.Repository
+	Validate     *validator.Validate
+	MaximumCards int
+	Rand         RandomService
+}
+
+func (s *service) MakeRandomCards(gameID string, numCards int) (game *Game, err error) {
+	// Makes random cards
+	existingGame, err := s.Repo.GetGame(gameID)
+
+	cards := make([]repository.Card, numCards)
+	words := make(WordList, numCards)
+
+	for i := range cards {
+		// Keep trying random words until a unique one is generated
+		for {
+			word, err := s.Rand.GetRandomNoun()
+			if err != nil {
+				log.Printf("error getting random noun: %v", err)
+				return &Game{}, err
+			}
+			if !words.contains(word) {
+				words[i] = word
+				break
+			}
+		}
+
+		if err != nil {
```

Comment:
Small nit - it looks like we're already handling the `s.Rand.GetRandomNoun` error on line 60 so it seems like we might not need to check it again here

Reply:
Nice catch